src/components/Nav.astro

```diff
@@ -17,7 +17,30 @@ const {input: MenuInput, label: MenuButton } = XElement
 
 const { navItems, pathname } = Astro.props as Props
 
-console.log(pathname, navItems)
+interface MenuInputElement extends HTMLInputElement {
+
+}
+
+const initMenuInput = (element: MenuInputElement) => {
+  const hamburgerElem = document.querySelector('[for="toggle"]')
+    element.onchange = () => {
+      element.checked
+        ? hamburgerElem.classList.add('is-active')
+        : hamburgerElem.classList.remove('is-active')
+    }
+
+    let innerWidth = window.innerWidth
+    window.addEventListener('resize', () => {
+      if (innerWidth < 768 && window.innerWidth >= 768) {
+        element.checked = false
+        hamburgerElem.classList.remove('is-active')
+      }
+
+      innerWidth = window.innerWidth
+    })
```

Comment:
```suggestion
    matchMedia('(min-width: 768px)').onchange = (event) => {
	  if (event.matches) {
	    hamburgerElem.classList.remove('is-active')
      }
    }
```

Reply:
Nice, thanks!  I always forget about the fancy `matchMedia` API